faiss/gpu/StandardGpuResources.h

```diff
@@ -183,10 +207,13 @@ class StandardGpuResources : public GpuResourcesProvider {
     /// Export a description of memory used for Python
     std::map<int, std::map<std::string, std::pair<int, size_t>>> getMemoryInfo()
             const;
-
     /// Returns the current default stream
     cudaStream_t getDefaultStream(int device);
 
+    /// Returns the raft handle for the given device which can be used to
+    /// make calls to other raft primitives.
+    raft::handle_t& getRaftHandle(int device);
```

Comment:
What's inside a `raft::handle_t`? Is it bad to copy by value (versus passing a reference here)?

Reply:
It's generally preferable to pass the same handle around (by reference). We've always passed it around that way, anyways within RAFT. Though with some of the recent changes it *may* be able to be passed around by value without causing excess copyies of the underlying resources.